software/controller/lib/core/commfail_alarm.h

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef COMMFAIL_ALARM_H
+#define COMMFAIL_ALARM_H
+
+#include <stdint.h>
+
+enum class AlarmProirity : uint8_t {
+  NONE = 0x00,
+  ALARM_LOW_PRIORITY = 0x01,
+  ALARM_MEDIUM_PRIORITY = 0x02,
+  ALARM_HIGH_PRIORITY = 0x03,
+};
+
+// Represents one of the stepper motors in the system
+class CommFailAlarm {
+public:
+  // This module monitors if any alarms are triggered
+
+  CommFailAlarm();
```

Comment:
you could just do `= default;` here to avoid defining the function in `cpp`, but actually I think it can be completely omitted because the compiler gives you one of these for free.

Reply:
removed as it is not needed & will be created by compiler